platform/darwin/src/MGLRasterSource.mm

```diff
@@ -37,76 +33,32 @@ - (instancetype)initWithIdentifier:(NSString *)identifier configurationURL:(NSUR
 }
 
 - (instancetype)initWithIdentifier:(NSString *)identifier configurationURL:(NSURL *)configurationURL tileSize:(CGFloat)tileSize {
-    if (self = [super initWithIdentifier:identifier configurationURL:configurationURL]) {
-        auto source = std::make_unique<mbgl::style::RasterSource>(identifier.UTF8String,
-                                                                  configurationURL.mgl_URLByStandardizingScheme.absoluteString.UTF8String,
-                                                                  uint16_t(round(tileSize)));
-        _pendingSource = std::move(source);
-        self.rawSource = _pendingSource.get();
-    }
-    return self;
+    auto source = std::make_unique<mbgl::style::RasterSource>(identifier.UTF8String,
+                                                              configurationURL.mgl_URLByStandardizingScheme.absoluteString.UTF8String,
+                                                              uint16_t(round(tileSize)));
+    return [super initWithPendingSource:std::move(source)];
```

Comment:
Initializers should always follow [the pattern](https://developer.apple.com/library/content/documentation/General/Conceptual/CocoaEncyclopedia/Initialization/Initialization.html#//apple_ref/doc/uid/TP40010810-CH6-SW58) `if (self = [super …]) { … } return self;`, even if it looks redundant.